internal/log/logrwrapper.go

```diff
@@ -0,0 +1,47 @@
+package log
+
+import (
+	"fmt"
+
+	"github.com/go-logr/logr"
+)
+
+// LogrWrapper is a nasty hack for turning the logr.Logger we get from NewLogger()
+// into something that go-control-plane can accept.
+// It seems pretty silly to take a zap logger, which is levelled, turn it into
+// a V-style logr Logger, then turn it back again with this, but here we are.
+// TODO(youngnick): Reopen the logging library discussion then do something about this.
+type LogrWrapper struct {
+	logr logr.Logger
+}
```

Comment:
I wouldn't have this in a separate package or be exported; that only serves to encourage using it instead of the preferred logging abstraction (logr). IMO this should be a private implementation detail of `internal/xds` (or `internal/xds/cache`?).